src/core/qgsvectorlayer.h

```diff
@@ -1621,6 +1621,35 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     bool readExtentFromXml() const;
 
+    /**
+     * Connects provider notification to triggerRepaint
+     *
+     * \since QGIS 3.0
+     */
+    void setRefreshOnNotifyEnabled( bool enabled );
```

Comment:
I'm wondering - should these changes be moved from QgsVectorLayer to QgsMapLayer? Potentially a raster data provider could also have some form of notification signal (such as when the file changes on disk).

Reply:
You are right. BTW, if not too expensive, the "watch file" could be enabled by default.